Pass gossipsub params to go-waku node

bacv opened the pull request:
Changes for passing gossipsub params to go-waku node. Related to waku-org/go-waku#489.
**This needs to be updated with the correct vendor commit**, after go-waku changes are merged.

danielSanchezQ commented:
This looks good. Update the vendor commit when available and check if logging errors persist. Do not know what happened on previous release.

bacv commented:
Updated vendor commit to match go-waku main branch.
Tested the logfix, added testcases for to verify:
```bash
cargo test loglevel_error -- --nocapture # no info logs
cargo test loglevel_info  -- --nocapture # info logs and levels above it
```
